src/providers/elixir.rs

```diff
@@ -32,27 +32,47 @@ impl Provider for ElixirProvider {
         plan.add_phase(setup);
 
         // Install Phase
-        let mut install_phase = Phase::install(Some("mix local.hex --force".to_string()));
-        install_phase.add_cmd("mix local.rebar --force");
+        let mut install_phase =
+            Phase::install(Some("MIX_ENV=prod mix local.hex --force".to_string()));
+        install_phase.only_include_files = Some(vec![
+            "config/config.exs".to_string(),
+            "config/prod.exs".to_string(),
+            "mix.exs".to_string(),
+            "mix.lock".to_string(),
+        ]);
+        install_phase.add_cmd("MIX_ENV=prod mix local.rebar --force");
         install_phase.add_cmd("mix deps.get --only prod");
         plan.add_phase(install_phase);
 
         // Build Phase
         let mut build_phase = Phase::build(Some("mix compile".to_string()));
+        build_phase.only_include_files = Some(vec![
+            "config/runtime.exs".to_string(),
+            "assets".to_string(),
+            "priv".to_string(),
+            "config".to_string(),
+            "lib".to_string(),
+        ]);
         let mix_exs_content = app.read_file("mix.exs")?;
-
         if mix_exs_content.contains("assets.deploy") {
             build_phase.add_cmd("mix assets.deploy".to_string());
         }
 
-        if mix_exs_content.contains("postgrex") && mix_exs_content.contains("ecto") {
```

Comment:
this won't work because the DATABASE_URL isn't available at this time. [Railway's own docs](https://docs.railway.com/guides/phoenix#prepare-our-phoenix-app-for-deployment) suggesting adding `mix ecto.setup` to the start phase instead.